### PR TITLE
Hook up to new Buy Now Echo config, make MP query

### PR DESCRIPTION
**Pull request by ashfurrow**
This PR does a few things:

- It updates Echo.
- It uses a newly added flag in Echo to enable Buy Now for our users remotely.
- It adds an admin flag to allow Artsy staff to enable the Buy Now flow immediately.
- It adds a metaphysics mutation to create and order and parse its order ID.
- It routes the user to a new route defined in Echo, with that order ID.

Regarding that last point, it is apparently not possible to tell JLRoutes "I want to go to _this_ route but use _this_ in place of `:id`", so I did a string replacement. Let me know if you have better ideas.

This PR is still a work-in-progress because:

- [x] ~We need to define how error-handling should work~ (update: ticket [is here](https://artsyproduct.atlassian.net/browse/PURCHASE-424)).
- [x] Verify it works on staging.
- [x] I need to write some tests.

This PR fixes [PURCHASE-388](https://artsyproduct.atlassian.net/browse/PURCHASE-388) and [PURCHASE-389](https://artsyproduct.atlassian.net/browse/PURCHASE-389). I also opened [PURCHASE-425](https://artsyproduct.atlassian.net/browse/PURCHASE-425) to address the spinner state.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by ashfurrow** [comment]
Okay, testing this locally is blocked by Force/Reaction having an old schema. I've opened a PR here to update them, but it's still a WIP itself: https://github.com/artsy/reaction/pull/1193

**Comment by ashfurrow** [comment]
Okay so here's what this looks like in the admin menu:

screen shot 2018-08-29 at 2 02 40 pm

This setting circumvents the Echo config, so we can test without enabling the feature for everyone. 

Right now, this routes to `/order2/:id`, but that's configurable through Echo too. However, the infrastructure for redirecting `/order2/:id` to `/order2/:id/shipping` hasn't been set up yet, so it'll 500 if you try it right now. Work to do that is ticketed in [PURCHASE-418](
https://artsyproduct.atlassian.net/browse/PURCHASE-418).

I'll wrap this PR up with some unit tests for the switchboard changes.

File: Artsy_Tests/View_Controller_Tests/Artwork/ARArtworkViewControllerBuyButtonTests.m [hunk 1]
```diff
@@ -31,94 +34,194 @@ - (void)tappedContactGallery;
...
+//            [[vcMock expect] presentErrorMessage:OCMOCK_ANY];
+//
+//            [[[[routerMock expect] andForwardToRealObject] classMethod] newBuyNowRequestWithArtworkID:@"artwork-id"];
+//
+//            [vc tappedBuyButton];
+//
+//            [routerMock verify];
+//            [vcMock verify];
+//        });
+//    });
```
Comment (ashfurrow) [inline Artsy_Tests/View_Controller_Tests/Artwork/ARArtworkViewControllerBuyButtonTests.m]: Haha oops.

**Pull request merged**